src/agents/OIDCCredentialAgent.js

```diff
@@ -1,5 +1,8 @@
+import Mutex from 'fast-mutex';
 import fetch from '../fetch';
 
+const MUTEX_LOCK_KEY = '@@TASKCLUSTER_SESSION';
```

Comment:
Maybe `taskcluster-client-web-credentials`?  Just to scope it to the name of the library..

Reply:
Sounds good.